### PR TITLE
347800043: Migrated cups_ipp construct-based parsers to use dtfabric #1893

### PR DESCRIPTION
**Pull request by joachimmetz**
[Code review: 347800043: Migrated cups_ipp construct-based parsers to use dtfabric #1893](https://codereview.appspot.com/347800043/)

### CODE REVIEW

File: plaso/parsers/cups_ipp.py [hunk 1]
```diff
@@ -214,123 +139,162 @@ def _GetStringValue(self, data_dict, name, default_value=None):
...
-        self.TYPE_GENERAL_INTEGER, self.TYPE_INTEGER, self.TYPE_ENUMERATION):
-      value_structure = self.INTEGER
-    elif type_id == self.TYPE_BOOL:
-      value_structure = self.BOOLEAN
-    elif type_id == self.TYPE_DATETIME:
-      value_structure = self.DATETIME
-    else:
-      value_structure = self.TEXT
+    value = None
+    if attribute.tag_value in (0x21, 0x23):
```
Comment (Onager) [inline plaso/parsers/cups_ipp.py]: Please extract these tag value constants out, this is very hard to follow as is.

**Comment by joachimmetz** [comment]
@Onager PTAL

File: plaso/parsers/cups_ipp.py [hunk 2]
```diff
@@ -87,8 +87,52 @@ class CupsIppParser(dtfabric_parser.DtFabricBaseParser):
...
+
+  _ASCII_STRING_VALUES = frozenset([
+      _TAG_VALUE_KEYWORD,
+      _TAG_VALUE_URI,
+      _TAG_VALUE_URI_SCHEME,
+      _TAG_VALUE_CHARSET,
+      _TAG_VALUE_NATURAL_LANGUAGE,
+      _TAG_VALUE_MEDIA_TYPE])
+
+  _INTEGER_TAG_VALUES = frozenset([
```
Comment (Onager) [inline plaso/parsers/cups_ipp.py]: Remove this, already defined above

File: plaso/parsers/cups_ipp.py [hunk 3]
```diff
@@ -162,36 +206,24 @@ def _ParseAttribute(self, file_object):
           'Unable to parse attribute with error: {0!s}'.format(exception))
 
     value = None
```
Comment (Onager) [inline plaso/parsers/cups_ipp.py]: This isn't needed, there's an else clause that defaults this to None anyway

File: plaso/parsers/cups_ipp.py (hunk 2 above)
Comment (joachimmetz) [reply plaso/parsers/cups_ipp.py]: ack
